covid19_sfbayarea/data/marin_scraper.py

```diff
@@ -0,0 +1,303 @@
+            "american indian/alaska native": "Native_Amer", "native hawaiian/pacific islander": "Pacific_Islander", "white": "White", "asian": "Asian", "multi or other race": "Multi or Other Race"}
+            # "Multiple_Race", "Other" are not separate in this data set - they are one value under "Multi or Other Race"
+
+	c_race_eth = {}
+	d_race_eth = {}
+	
+	for row in csv_strs[1:]:
+		split = row.split(',')
+		race_eth = split[0].lower()
+		if race_eth not in key_mapping:
+			raise ValueError("The race_eth groups have changed.")
+		else:
+			c_race_eth[key_mapping[race_eth]] = int(split[2])
+			d_race_eth[key_mapping[race_eth]] = int(split[6])
+
+	return c_race_eth, d_race_eth
+
+def get_test_series(chart_id: str, url: str) -> Tuple:
+	"""This method gets the date, the number of positive and negative tests on that date, and the number of cumulative positive and negative tests."""
+
+	csv_ = extract_csvs(chart_id, url)
+	series = []
+
+	csv_strs = csv_.splitlines()
+	keys = csv_strs[0].split(',')
+	
+	test_history = []
+	
+	# Grab the dates, which are in the header
+	for entry in csv_strs[:1][0].split(',')[1:]: 
+		# need to exclude very first item in the csv_strs[:1][0].split(',') list (which is the value 'Date')
+		daily = {}
+		date_time_obj = datetime.strptime(entry, '%m/%d/%Y')
+		daily["date"] = date_time_obj.isoformat()
+		series.append(daily)
+
+	# The slicing makes this if statement hard to look at... there must be a better way?
+	if csv_strs[1:2][0].split(',')[:1][0] != 'Positive Tests' and csv_strs[2:][0].split(',')[:1][0] != 'Negative Tests':
+		raise ValueError('The kinds of tests have changed.')
+
+	# Grab the positive test result numbers, which is in the second row. 
+	# [1:] is included to make sure that 'Positive Tests' is not captured.
+	p_entries = csv_strs[1:2][0].split(',')[1:]
+	n_entries = csv_strs[2:][0].split(',')[1:]
+	
+	get_test_series_helper(series, p_entries, ['positive', 'cumul_pos'])
+	get_test_series_helper(series, n_entries, ['negative', 'cumul_neg'])
+	
+	return series   
+
+def get_test_series_helper(series: list, entries: list, keys: list) -> List:
+	"""This method helps get the pos/neg test count and the cumulative pos/neg test count."""
+	
+	# initialize values cumulative number, the positive/negative and cumul_pos/neg values for the first day, and the index needed for the while loop.
+	
+	# there's probably a more efficient way to do all of this, but I just wasn't sure.
+	cumul = int(entries[0])
+	series[0][keys[0]] = int(entries[0])
+	series[0][keys[1]] = cumul
+	index = 1	
+
+	while index < len(series):
+		# get a particular day
+		day = series[index]
+		curr = int(entries[index])
+		# get pos/neg test count
+		day[keys[0]] = int(curr)
+		# add that day's pos/neg test count to get cumulative number of positive tests
+		cumul += curr
+		day[keys[1]] = cumul 
+		index += 1
+	return series
+
+
+get_county_data()
```

Comment:
This needs to go, `get_county_data()` needs to be renamed and should return the dict instead of printing, and `covid19_sfbayarea/data/__init__.py` needs to list this module. :)

https://github.com/sfbrigade/data-covid19-sfbayarea/blob/cf49d5a6d3efe0ac43f93dcd0573c5e4b38c4dde/covid19_sfbayarea/data/__init__.py#L2-L15

That’s what’s required so this can run the same way the other scrapers do:

```console
$ bash run_scraper_data.sh marin
# Or:
$ python scraper_data.py marin

# Or to dump all counties into a directory:
$ bash run_scraper_data.sh --output county_data
# Or:
$ python scraper_data.py --output county_data
```

Reply:
thanks Rob! I'll rename the file at the end when this PR is ready to go (if I rename the file to marin.py now then these comments will be on a deleted file. Sorry about the inconvenience.

Reply:
thank you Rob!!